### DISCUSSION TITLE
Bundle with Rollup instead of browserify

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#259] [Development]
In general it doesn't really matter with what to build/bundle/transpile
But Rollup is very flexible and easy to use tool for it.

Ideally everything should be working without bundeling. However the major stumbling block is resolving path to node modules in the browser context. 

What are the requirements for bundle in Cordova?

**Comment by tobias-klein** [comment]
What's driving this change proposal? Are there issues with Browserify?

Regarding Cordova:
There are no specific bundling requirements for Cordova. If it works in a regular Browser it should also work in the Cordova webview.

**Reply by zhuiks** [reply]
No, there are no issues with Browserify that I'm aware of. Just wanted to add my 5 cents... :)

When I was experimenting with Material Web Components I needed something to resolve node modules for browser context. Rollup was mentioned in their documentation, so I went ahead and tried it. And I liked its simplicity and flexibility. Although in this case the end result was the same as could have been achieved with Browserify - a bundle.
Browserify have been around for a while. Rollup is a relatively new tool that tries to solve weak places of the bundlers/transpilers what ever they are.

I think we should stick with one tool. And for now I guess there is no real need in changing to something else.
So Browserify it is

Thanks for listening to my "5 cents" :smiley:

**Reply by tobias-klein** [reply]
Sure. Let's keep this in the back of our minds if we run into roadblocks with Browserify.
